src/ExternalSearch.Providers.VatLayer/VatLayerExternalSearchProvider.cs

```diff
@@ -374,6 +376,59 @@ public IPreviewImage GetPrimaryEntityPreviewImage(ExecutionContext context, IExt
             }
         }
 
+
+        public ConnectionVerificationResult VerifyConnection(ExecutionContext context, IReadOnlyDictionary<string, object> config)
+        {
+            IDictionary<string, object> configDict = config.ToDictionary(entry => entry.Key, entry => entry.Value);
+            var jobData = new VatLayerExternalSearchJobData(configDict);
+
+            var vat = WebUtility.UrlEncode("GB765970776");
+            var client = new RestClient("http://www.apilayer.net/api");
+            var request = new RestRequest($"validate?access_key={jobData.ApiToken}&vat_number={vat}&format=1", Method.GET);
+
+            var response = client.ExecuteAsync<VatLayerResponse>(request).Result;
+
+            return ConstructVerifyConnectionResponse(response);
+        }
+
+        private ConnectionVerificationResult ConstructVerifyConnectionResponse(IRestResponse<VatLayerResponse> response)
+        {
+            var isSuccessResponse = response.IsSuccessful;
+            var errorMessageBase = $"{Constants.ProviderName} returned \"{(int)response.StatusCode} {response.StatusDescription}\".";
+            if (response.ErrorException != null)
+            {
+                return new ConnectionVerificationResult(false, $"{errorMessageBase} {(!string.IsNullOrWhiteSpace(response.ErrorException.Message) ? response.ErrorException.Message : "This could be due to breaking changes in the external system")}.");
+            }
+
+            var responseData = response.Data;
+            if (responseData == null || (responseData != null && !responseData.Valid))
```

Comment:
Could this be simplified to `if (responseData?.Valid != true)`

Reply:
updated